iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java

```diff
@@ -172,32 +168,36 @@ static BitSetIntIterable fillVendors(BitReader bbv, FieldDefs maxVendor, FieldDe
      */
     static int vendorIdsFromRange(BitReader bbv, BitSet bs, int numberOfVendorEntriesOffset,
             Optional<FieldDefs> maxVendor) {
-        int numberOfVendorEntries = bbv.readBits12(numberOfVendorEntriesOffset);
+
+        final int numberOfVendorEntries = bbv.readBits12(numberOfVendorEntriesOffset);
+        final int maxV = maxVendor.map(maxVF -> bbv.readBits16(maxVF)).orElse(Integer.MAX_VALUE);
+        final int vendorIdFieldLength = FieldDefs.START_OR_ONLY_VENDOR_ID.getLength(bbv);
+        final int vendorIdMask = 0xFFFFFFFF >>> (32 - vendorIdFieldLength);
         int offset = numberOfVendorEntriesOffset + FieldDefs.NUM_ENTRIES.getLength(bbv);
-        int maxV = maxVendor.map(maxVF -> bbv.readBits16(maxVF)).orElse(Integer.MAX_VALUE);
 
         for (int j = 0; j < numberOfVendorEntries; j++) {
             boolean isRangeEntry = bbv.readBits1(offset++);
-            int startOrOnlyVendorId = bbv.readBits16(offset);
-            offset += FieldDefs.START_OR_ONLY_VENDOR_ID.getLength(bbv);
             if (isRangeEntry) {
-                int endVendorId = bbv.readBits16(offset);
-                offset += FieldDefs.START_OR_ONLY_VENDOR_ID.getLength(bbv);
+                final int content = bbv.readBits32(offset);
```

Comment:
there is significant code here to eliminate a call to readBits16. I don't think it's worth it unless you can show a benchmark that it adds value.